\section{Zebrafish as a Model Organism}

Laboratory model organisms have been a staple of research since the dawn of the scientific endeavor, but only in the past century has standardization of models allowed for improvements in reproducibility and reliability among experiments. One model in particular, the C57BL/6 \textit{Mus musculus} mouse model, has become a ubiquitous feature of every major research institution all over the world due to their clonal nature\footnote{Genetic diversity between individual C57BL/6 mice is in the range of x single nucleotide polymorphisms per individual in a genome of x bases (ref).}, relative ease of use, and minimal expense\footnote{A single C57BL/6 mouse from Jackson Laboratories (jax.org) at the time of writing is \$24 USD.}. However, their genetic homogeneity fails to reproduce many phenotypes seen in human disease, making them an excellent model for some disorders and an insufficient one for others. This is nowhere more true than in developmental biology. Although mouse viviparous development is extremely well defined and stereotyped over the course of gestation, that is precisely the challenge. Gestation is an internal and ongoing process of physiological and anatomical development and while it is possible to catalog the process of development in snapshots in time through vivisection, it is impossible to understand the kinetics and processes of development using a model that does not allow for immediate visual accessibility. 

In the 1980s this led researchers in Oregon to seek a model that would allow for the full visual access only possible in oviparous organisms. Although \textit{Xenopus} frogs had been in use for some time, their long time to sexual maturity (up to 2 years for \textit{Xenopus laevis}, the dominant model at the time) and other challenges led researchers to a fish model, at the root of the land-adapted branch of the tree of life. A happenstance purchase at a local pet store led to the establishment of the imminently powerful zebrafish model, which led to seminal and otherwise impossible findings in developmental biology. This model has since found applications in nearly every field of biology for many of the same reasons: optical transparency, extremely rapid development, high fecundity, and genetic tractability. These features make the zebrafish a potent and robust tool for the study of many different biological processes and, thanks to their intolerance for inbreeding, have remained a genetically diverse outbred\footnote{The scale of zebrafish outbreeding is difficult to define, even among strains that are used in research laboratories. For instance, the majority of the work in subsequent chapters is done in the *AB background, a classic wild-type reference strain used around the world. This strain, similar to other strains, has upwards of 6000 copy number variations between individuals (~15\% of the genome) in addition to approximately 1 single nucleotide polymorphism (SNP) for every 500 bases of genome sequence. Experimentalist anecdotes of the intolerance of the zebrafish for inbreeding are ubiquitous, as this widespread genome-level heterozygosis appears to confer some important advantages to individuals, especially in regard to fertility.} model for research that allows for more sophisticated modeling of complex processes with the caveat that it also fuels a need for high n-values due to inherent variation between individuals. Conversely, detectable effects in a high-noise environment are often more robust associations than those found in a more monoclonal environment.

\textit{Danio rerio} is a small (1-2 cm in length) freshwater fish natively found in the Ganges River and tribituaries in India. A robust fish in the pet trade, they possess reflective stripes that play a role in predator evasion, as well as contributing to their mating cycle. They spawn freely, releasing individual eggs into the water where they can be fertilized by an engaged male\footnote{Or some bystander male.}. The larvae consume waterborne microorganisms for food as they undergo metamorphosis, changing key elements of the body plan to develop the adult anatomy. 

Only in the past twenty years has an earnest effort been put forth to develop the zebrafish as a model for immunological studies. Although it has long been known that zebrafish, like all vertebrates, possess the full repertoire of immune cells and responses, little was done with that knowledge until recently, given the perceived benefits of the C57BL/6 model, which more closely resembles some aspects of the human immune system and has a superabundance of useful genetic tools with which to study immune responses in cancer, inflammation, autoimmunity, and infection. 
The mouse has served as the model for immunology for the past 50 years. It has enabled monumental discoveries that have resulted in new medications and therapies to treat nearly every conceivable human disease and is the foundation of every single chemotherapeutic medicine on the market today. The diminutive mouse is an outstanding model for a vast array of human diseases and continues to be the go-to model for many processes and disorders. However, classical inbred mouse models, including C57BL/6 and other popular lines, including BALB/c, A/J, and 129S1, fail to replicate defining characteristics of tuberculosis in ways that compromise our ability to apply findings from these models to the kinetics and pathology of human disease. For instance, the C57BL/6 mouse is highly resistant to acute tuberculosis disease; these mice can be infected with standard laboratory strains of \textit{M. tuberculosis} and succumb approximately 300 days later\footnote{Bacterial strain variations and dosage can alter this somewhat, but infection with the reference strain H37Rv will result in a highly reproducible and sudden series of death almost a year after initial infection.}. 

\subsection{Zebrafish Husbandry}

Zebrafish husbandry serves as one of the major advantages of the model. Social fish, they can be maintained at a reasonably high density per unit volume as adults\footnote{Approximately 250 mL of water per fish is more than sufficient for rapid growth to adulthood and breeding success.}, which can then spawn ~200 eggs per week per female. Our fish are kept on a constantly recirculating rack system from Aquaneering, which replaces 90\%+ the volume of each of ~1,000 tanks once per hour, 24 hours a day. The fish are diurnal animals and photoperiodic breeders, which greatly eases experimenter access, as they will reproduce reliably only in the morning, which can be artificially set to the convenience of the laboratory. The fish are kept at 28.5$^{\circ}$C on a 14hr-10hr light-dark cycle and kept in either 3 or 6 L tanks. Reverse osmosis water is maintained at 600-700 \textmu S conductivity by addition of Instant Ocean Sea Salt (\#SS15-10) and a pH between 7.0 and 7.4 (buffered by automated addition of sodium bicarbonate; Arm \& Hammer Pure Baking Soda [\#426292]). All of this work were performed in accordance and compliance with policies approved by the Duke University Institutional Animal Care and Use Committee (protocol \#A091-20-04).

Infected adult zebrafish are maintained at an identical 14hr-10hr light cycle at 28.5$^{\circ}$C in an isolated incubator (ThermoFisher \#PR505755L) physically separated from the primary fish system. Fish are kept at no greater than 1 fish/100 mL of water in Aquaneering crossing cages (Aquaneering \#ZHCT100) and are fed daily with the standard fish diet in our lab (Skretting \#GEMMA Micro 500). Water is changed daily using water taken from the primary fish system. While sex is not a standard factor we account for in the analysis of our experiments, approximately equal numbers of fish of each sex are used for infection experiments. Any adult fish exhibiting signs of imminent distress or morbidity\footnote{inability to right, flared scales, labored breathing, obvious open wounds} are euthanized. Standard adult experiments are conducted for a minimum of 14 days, although infection will continue to progress over the course of approximately four to six weeks if needed. Fish are humanely euthanized at the conclusion of experimentation by tricaine overdose and decapitation.

For infection, fish were anesthetized in 120 \textmu g/mL tricaine. Single-use aliquots of single cell suspensions of \textit{M. marinum} were thawed and diluted in sterile PBS and zebrafish were injected with 10 \textmu L of a solution containing 200-1000\footnote{While the precise number is kept consistent within each experiment and subsequent replicates, different initial inocula are chosen based on the desired kinetics of disease. Lower initial doses are likely to be superior for identifying factors regulating bacterial control while higher doses are better if the desired outcome is to study aspects of granuloma biology, as high starting doses tend to result in a greater number of granulomas forming earlier in disease. This often appears to balance out as the disease progresses and some functional upper limit on bacillary load is reached.} fluorescent bacteria using a back-loaded insulin syringe (BD \#08290-3284-38). Injection is done into the peritoneal cavity of the fish, acheiving a systemic infection within the cavity and which spreads to all major organ systems\footnote{Although brain involvement is only rarely seen. It is not a standard tissue for us to analyze in the lab, but previous analysis has generally failed to identify meningitis as a manifestation of \textit{M. marinum} infection in the zebrafish.}. 

The larval zebrafish, on account of the high fecundity of the model, is a key tool in these studies. We can readily collect >1,000 embryos at once and can reasonably infect ~200 per hour, allowing for high n-value screening approaches. This, in addition to their remarkable ability to recover and highly stereotyped developmental patterning, facilitates reproducible studies in many fields of biology including the work described herein. Embryos are collected in the early- to mid-afternoon, sterilized in 0.05\% hypochlorite solution, and allowed to develop in the presence of ~0.001\% methylene blue in E3 medium (5 mM NaCl (Fisher Scientific \#S271), 178 \textmu M KCl (VWR \#BDH9258), 328 \textmu M CaCl2 (VWR \#BDH9224), 400 \textmu M MgCl2 (Ward'Äôs Scientific \#470301)) at no more than 150 larvae per dish. At 24 hours post fertilization, they are transferred into E3 supplemented with 1-phenyl-2-thiourea (PTU, Sigma-Aldrich \#P7629) at a final concentration of 45 \textmu g/mL to prevent melanization if they are to be used for imaging studies. Otherwise, they are allowed to develop in unmodified E3 medium until 3-4 days post fertilization, when they are put into the nursery system to be raised to adulthood. Infected larval zebrafish are euthanized prior to 8 days post fertilization in all instances. Zebrafish are of indeterminate sex\footnote{This is only strictly true for laboratory strains of zebrafish. Wild zebrafish have a sex determining region on chromosome 4 that has been repeatedly lost under laboratory culture conditions, for reasons that remain mostly unknown. In the lab, a variety of factors seem to influence the eventual sex determination of the fish, including rearing density and caloric allotment, in addition to various genetic factors. Despite the lack of strict genetic determinants of sex, the standard 1:1 female:male ratio is generally maintained under normal conditions.} until they reach the juvenile stage of development, so no distinctions are or can be made on the basis of sex in larval zebrafish studies.

For ease of manipulation and the minimization of distress, larval zebrafish are anesthetized in approximately 160 \textmu g/mL of tricaine (MS-222 or Tricaine-S\footnote{Now known as "Syncaine"}, Syndel \#ANADA 200-226) prior to injection. Injection of \textit{M. marinum} is done by injecting 2 days post fertilization larvae into a developmentally undefined peri-notochordal space between the somitic muscle layers along the trunk. Approximately 50-150 fluorescent bacteria are injected, spreading allong the anterior-posterior length of the fish and establishing a largely localized infection along the avascular trunk. This infection location allows for facile quantitation of vascular aberrations, an essential prerequisite for this study. Injection of TDM is conducted similary. Trehalose 6-6'-dimycolate from \textit{Mycobacterium bovis} (TDM, Sigma-Aldrich \#T3034) was resuspended in 2:1 v/v chloroform:methanol at 1 mg/mL and stored at -80$^{\circ}$C. Prior to use, the liquid is evaporated under vacuum and emulsified in incomplete Freund's adjuvant (IFA, Sigma-Aldrich \#F5506) at 2 mg/mL. Larvae are then anesthetized and injected with approximately 10-20 nL of TDM/IFA or IFA along the trunk in this same undefined peri-notochordal space. The droplets coalesce into spheres within 10-15 minutes and remain in place for the entire experimental duration. Larvae are allowed to recover in E3 medium supplemented with PTU and raised in a 28.5$^{\circ}$C incubator.


\subsection{Zebrafish and their History in Developmental Biology}
\subsection{Modern Applications of Laboratory Zebrafish}
\subsection{Zebrafish as a Model System for the Study of Immunity}
\subsection{Challenges in the Use of the Zebrafish Model}
\section{Mycobacterium marinum-Zebrafish Model of Tuberculosis}

Other popular laboratory models of tuberculosis are able to form granulomas, including rabbits and guinea pigs; the former is highly resistant to tuberculosis while the latter is highly susceptible. However, these tend to require maintenance via outbreeding, are larger mammals with associated higher husbandry costs, and are devoid of most useful genetic tools. This left a clear gap in our ability to understand some of the aspects of this important human disease that required innovative new approaches and a whole new paradigm. 

A foundational study in 2002 set the tone for the next two decades of research into host-microbe interactions in the zebrafish. Davis \& Ramakrishnan took advantage of the optical transparency and manipulative amenability of the zebrafish larvae to infect them with an aquatic pathogen in the \textit{Mycobacterium} genus \textit{Mycobacterium marinum}. \textit{M. marinum} is a globally dispersed pathogen of fish and amphibians that causes tuberculosis in fish, which tends to manifest in superficial lesions, spinal deformities, and wasting. The use of this heterologous host-pathogen system allowed for the first ever \textit{in vivo} visualization of the early processes of granuloma formation through the interactions between the invading bacteria and the responding host macrophages, which serve as the first responding innate immune cells to mycobacterial infections. 

Further developments over the following years, most notably by Swain et al. in 2006, established the zebrafish as a sophisticated and multifaceted model that allows for both comprehensive live imaging of the early processes of infection and dissection of the later stages of infection using adult zebrafish that form granulomas morphologically similar to those formed by humans in response to both \textit{M. tuberculosis} and during opportunistic infections by \textit{M. marinum}. These findings set the stage for the continued development of the zebrafish-\textit{M. marinum} model of tuberculosis and has enabled the study of processes of human disease that have been long described but previously unable to be evaluated.

\subsection{Relevance and Natural History of \textit{Mycobacterium marinum}}
\subsection{Deficits of Mouse Models of Tuberculosis}

Historically, the C57BL/6 mouse model of tuberculosis has served as the gold standard for studies on host-pathogen interactions in tuberculosis and has been used to identify major host-protective factors as well as bacterial virulence factors. However, even among other mouse models, the C57BL/6 model has clear challenges in the field of tuberculosis biology. For one, C57BL/6 does not form necrotic granulomas under standard laboratory doses of laboratory strains of \textit{M. tuberculosis}. This discordance between the observed human phenotype and the mouse model leaves an abundance of room for misinterpretation of data that may or may not be translatable to the human disease context. The loci of infection in this mice becomes a granulocytic infiltrate, with abundance cellular involvement and engagement with the bacteria, including the presence of lymphocytes. One of the standard arguments that the granuloma has host-detrimental aspects is that the physical structure blocks T cell-mycobacteria interactions, which may be able to serve an important host-protective role. That this is incapable of being modeled in the C57BL/6 model leaves room for competing models to model these elements of granuloma biology.

More recently, the C3H/FeJ "Kramnik" mouse model has come into widespread use. This model is able to model granuloma biology, as ~50\% of infected mice will form granulomas. However, this is also a hypersusceptible model of infection, with the average mouse succumbing to infection within approximately 3 months. In humans, disease progression can be of either very rapid progression (length?) or develop in severity over the course of many years (citations). Thus, there remains an unmet need for a model that (a) forms granulomas and mirrors other important aspects of human disease and (b) exhibits a spectrum of disease presentation similar to that seen in humans. 


%%%


\section{Data and Code Availability}

All raw image quantitation data (as .csv files) along with R scripts used for analysis are publicly archived via Zenodo at doi://10.5281/zenodo.6816429. Where appropriate and feasible, minimally processed image files have been included. Raw image files are available upon request to the author due to their large file size. Additionally, any ImageJ macros (in .py format) used for analysis are included for user convenience.

Experimental Model and Subject Details

Zebrafish Husbandry



\section{Mycobacterium marinum}

\subsection{Culture Conditions}

All strains are derived from M. marinum strain M (ATCC \#BAA-535) (Ramakrishnan \& Falkow 1994). Hygromycin-resistant fluorescent strains expressing the tdTomato (Oehlers et al. 2015), mCerulean, or EBFP2 fluorescent proteins have been described previously (Takaki et al., 2013). Bacterial culture was carried out on either 7H10 agar (Difco \#262710) plates supplemented with Middlebrook OADC growth supplement (10\% v/v; Sigma-Aldrich \#M0678) and 50 \textmu g/mL Hygromycin B (ThermoFisher \#10687010) or liquid 7H9 media (Difco \#271310) supplemented with Middlebrook OADC growth supplement (10\% v/v), 0.05\% Tween 80 (Sigma-Aldrich \#P1754), and 50 \textmu g/mL Hygromycin B. 

Single cell preparations of these bacteria were prepared and stored as single-use aliquots at -80$^{\circ}$C. Briefly, bacteria were grown at 33$^{\circ}$C in 50 mL 7H9 supplemented with 10\% OADC (Sigma-Aldrich \#M0678), 0.05\% Tween-80 (Sigma-Aldrich \#P1754), and 50 \textmu g/ml hygromycin B (Invitrogen \#10687010) (7H9 Complete). Once cultures reach OD\textsubscript{600} 0.55-0.8, they are spun down at 4600 rcf for 15 minutes and resuspended in 5 mL PBS-T (1x PBS with 0.05\% tyloxapol (Sigma-Aldrich \#T8761)) and bring to 25 mL total in PBS-T. They are spun and washed 2x in 25 mL PBS-T each time and then resuspend in 2 mL of 7H9 with 10\% OADC (Freezing 7H9) and split into 250 \textmu L aliquots and homogenized 10x using a 1 mL syringe and 27G needle (BD \#309623). Next, a soft spin at 770 rcf for 1 minute is done to pellet larger clumps and the supernatants are collected and then push the pooled supernatants through a 5 \textmu m filter (Millipore \#SLSV025LS) using a 10 mL syringe. The suspension is collected in 1.5 mL microfuge tubes and spun at 10000 rcf for 5 minutes. Final resuspension of pellet is done in freezing 7H9 and aliquoted into single use aliquots and concentration is calculated by fluorescent bacteria on a hemocytometer and by colony forming units on selective media.

CFU Assays

Colony forming unit assays were conducted by complete homogenization of whole adult zebrafish after euthanasia by tricaine overdose and external cleansing of the skin using 70\% ethanol. A single 6.5 mm ceramic bead (Omni \#19-682) was added to in a pre-filled bead mill tube containing 2.8 mm stainless steel beads (Sigma-Aldrich \#Z763829-50EA) was homogenized on a bead mill (MP Bio \#116004500) for a single 25 second interval at 5 meters/second. Lysate was plated on 7H10 plates supplemented with 10\% OADC,  hygromycin B (50 \textmu g/mL), amphotericin B (Gibco \#15290-026) (10 \textmu g/mL), and polymyxin B (Cayman Chemical \#14157) (25 \textmu g/mL). Lysate was plated in serial 1:10 dilutions up to 10\textsuperscript{-5}. Cultures were grown for 10-14 days prior to counting visible colonies. Where possible (due to contamination inherent to the assay), confirmatory counting was performed at 21 days after plating to capture slow-growing colonies, which are quite rare in our hands. Plates displaying overt contamination that occluded colony growth were excluded from further analysis.



THP-1 Culture

THP-1 (ATCC TIB-202) cells are a human monocytic cell culture line derived from a 1-year-old 46XY patient suffering from acute myelogenous leukemia in 1980. These cells were sourced from the Duke Cell Culture Facility and tested for mycoplasma prior to receipt. Cells are cultured in RPMI-1640 (Sigma-Aldrich \#R8758) supplemented with glucose (Sigma-Aldrich \#G8769), HEPES (Gibco \#15630), sodium pyruvate (Gibco \#11360) and 10\% non-heat inactivated FBS (Sigma-Aldrich \#F2442) in T-75 flasks (CellStar \#658170) in a 37$^{\circ}$C incubator with 5\% CO\textsubscript{2}. Cells were cultured for no greater than 10-12 passages prior to use and were then disposed of.

Method Details




Establishment of Transgenic Lines

Transgenic lines were established using tol2 transgenesis via the tol2kit (Kwan et al., 2007) and constructed by Gateway cloning. 

The p5e \textit{irg1} construct was generated by restriction digestion of irg1-pTol2linkerswitch (Sanderson et al., 2015) (a gift from Christopher Hall) with FseI and XmaI and then blunted using T4 DNA polymerase (NEB \#M0203S) per the manufacturer'Äôs instructions. Simultaneously, p5e MCS (Kwan et al., 2007) PCR linearized using inverted T3 and T7 promoter primers (\seqsplit{5'Äô- CCCTATAGTGAGTCGTATTAC-3'}, \seqsplit{5'Äô- TCCCTTTAGTGAGGGTTAAT-3'Äô}), digested with DpnI and PCR purified. These fragments were then ligated using T4 DNA ligase (NEB \#M0202S) to generate p5e irg1. This plasmid was then recombined with pME tdTomato (Addgene \#135202), p3e ubb pA (Addgene \#188702), and pDEST tol2 ubb pA (Addgene \#188701) by Gateway cloning (ThermoFisher \#12538120) to generate the pTol2 irg1:tdTomato construct that was then injected into single cell embryos alongside 15 ng/\textmu L tol2 mRNA (Balciunas et al., 2006) in 1x Tango buffer (ThermoScientific \#BY5). Candidate founders were selected based on fluorescence at 3 dpf, raised to adulthood, and outcrossed to *AB to establish the line, which transmits at ~50\% frequency, suggesting a single insertion locus and has exhibited stable expression over ~6 generations.

Tg(irg1:VIVIT-tdTomato\textsuperscript{xt38}), in which the inhibitory peptide VIVIT conjugated to the fluorescent protein tdTomato is expressed strictly in macrophages, was constructed by recombination of p5E irg1 (Addgene \#188698), pME VIVIT NS (Addgene \#188699), p3E tdTomato (Addgene \#188700), and pDEST tol2 Ubb pA (Addgene \#188701). Reactions were incubated at equimolar ratios overnight in a 25$^{\circ}$C thermocycler with heated lid, with volumes calculated using the ÄúLR Ratios Calculatorù Excel document. The irg1 promoter was first described by Sanderson et al. as a macrophage-specific inducible promoter, but our lab has found that this element often drives basal expression in macrophages as well, likely in an insertion-site-dependent manner. Tg(irg1:tdTomato\textsuperscript{xt40}) was similarly generated by recombination of p5E irg1, pME tdTomato, p3E Ubb pA, and pDEST tol2 Ubb pA. 

The middle element, pME VIVIT NS was constructed by a synthetic templated PCR after annealing. Two oligonucleotides from Integrated DNA Technologies (IDT) were annealed by heating to 95$^{\circ}$C and then slowly cooled to room temperature (sense: \seqsplit{5'Äô-GCCATCATGGCAGGACCACACCCGGTGATTGTTATCACTGGACCACATGAGGAG-3'}Äô, anti-sense: \seqsplit{5'-CTCCTCATGTGGTCCAGTGATAACAATCACCGGGTGTGGTCCTGCCATGATGGC-3'Äô}). This was then used as a template for PCR using two primers to add the \textit{attB1} and \textit{attB2} sites required for Gateway recombination into pDONR 221 (forward: \seqsplit{5'-GGGGACAAGTTTGTACAAAAAAGCAGGCTGCCATCATGGCAGGACC-3'Äô}, reverse: \seqsplit{5'Äô-GGGGACCACTTTGTACAAGAAAGCTGGGTACTCCTCATGTGGTCCAGTG-3'}Äô). This PCR product was then column purified and recombined into pDONR 221 (reference) using BP Clonase II (ThermoFisher \#11789020) to generate pME VIVIT NS (no stop) (Addgene \#188699). Constructs were verified by either Sanger sequencing or whole plasmid sequencing from Plasmidsaurus and have been submitted to Addgene, which provides additional whole plasmid sequencing verification.

Genotyping to differentiate the irg1:tdTomato\textsuperscript{xt40} and irg1:VIVIT-tdTomato\textsuperscript{xt38} lines can be performed where necessary (either for intentional experimental blinding or due to incidental mixing of fish during husbandry or experimentation) by PCR and gel electrophoresis. Primers (\seqsplit{5'-GATTTAGGTGACACTATAGATTCAGAGCTCGCACAGG-3'Äô}, \seqsplit{5'Äô-ATCTCGAACTCGTGGCC-3'Ä}ô) amplify across the 3'Äô end of the \textit{irg1} promoter and into the 5' end of the tdTomato insert. VIVIT+ fish display a 236 bp band while tdTomato-only fish display a 163 bp band. No band is seen in sibling fish lacking an \textit{irg1} transgene. 

Mutation via CRISPR/Cas9

Generation of mutants in card9, nfatc2a, and nfatc3a was performed as described previously. Alleles were identified by outcrossing of mosaic adults to wildtype *AB and Sanger sequencing of F1 adults. DNA extraction was conducted by cellular lysis in 50 mM sodium hydroxide as described previously (Meeker et al., 2007). Briefly, either adult zebrafish tail fins or whole larvae were collected in 50 mM NaOH in H\textsubscript{2}O and lysed at 98$^{\circ}$C for 12 minutes in a thermocycler and then neutralized by 1:10 addition a solution of 1M Tris-HCl (pH 8) in 10x TE (100 mM Tris, 10 mM EDTA). This solution was then directly used as the template for downstream PCR reactions. 

The allele card9\textsuperscript{xt31} was generated by injection of a single guide RNA into single-cell embryos (guide sequence: \seqsplit{5'Äô-TAATACGACTCACTATAGGGCAAGGTGCTGAGCAGCGGTTTTAGAGCTAGAA-3'}Äô). We identified an allele containing a 28 bp insertion, resulting in an immediate downstream frameshift leading to a premature termination codon at amino acid 59 (with missense mutations beginning at amino acid 47). Genotyping was performed using high-resolution melt analysis (HRMA) using the MeltDoctor Master Mix  (Applied Biosystems \#4415450) with primers flanking the sgRNA site (\seqsplit{5'Äô-CCTTATCTGAGACAGTGCAAGGTGC-3'}Äô, \seqsplit{5'Äô-TTACCAACTTTGCGGCGTCTG-3'Äô}). Amplification for Sanger sequencing was performed using primers (\seqsplit{5'-GTTTTCCCAGTCACGACCGAATGCTTCTCATCAAGACC-3'Äô}, \seqsplit{5'Äô-CGAATGCTTCTCATCAAGACC-3'Äô}.

The allele nfatc2a\textsuperscript{xt69} was generated by simultaneous injection of two neighboring guide RNAs to increase odds of a larger intervening deletion (guide sequences: \seqsplit{5'Äô-TAATACGACTCACTATAGGGCTGCGAGAACGGGCCACGTTTTAGAGCTAGAA-3'}Äô, \seqsplit{5'Äô-TAATACGACTCACTATAGGCAGCCCGTCGCCCCACGGGTTTTAGAGCTAGAA-3'}). We identified a mutation consisting of a complex, bipartite insertion/deletion leading to a net 4 bp insertion and frameshift leading to a premature termination codon at amino acid 272 (of 894, prior to the DNA binding domain). Genotyping can be performed by one of two distinct restriction digest-based methods. The original method was performed by restriction digest of the ~500 bp PCR product produced by the listed sequencing primers (\seqsplit{5'Äô-TAGAAGGCACAGTCGAGGCTCGAGGCTTTCTGGAGACCTCTGTCC-3'}, \seqsplit{5'-TGACACACATTCCACAGGGTCTCTAGAGGTTTGCCCTTCATATCCTGC-3'}Äô, underlined portion base pairs with the genomic sequence); digestion was with PflMI (NEB \#R0509) directly in the PCR reaction mixture. PCR was performed using LongAmp Taq (NEB \#M0323) strictly for reasons of buffer compatibility with the restriction enzyme. Digestion was carried out for ~3 hours at 37$^{\circ}$ in the presence of rSAP (NEB \#M0371) to minimize background. Sanger sequencing was conducted on undigested PCR products using the vendor (Eton Biosciences) supplied "ÄúBGH Reverse"Äù primer (\seqsplit{5'-TAGAAGGCACAGTCGAGG-3'Äô}) corresponding to the appended 5'Äô tail of the forward PCR primer. 

The second method utilizes a separate set of primers (\seqsplit{5'-CCTCTATGCAAACGCACCTACG-3'}Äô, \seqsplit{5'Äô-GTGATGCTCCTTGTGGCCAC-3'Ä}ô) to generate a 102-106 bp PCR product spanning the mutation site. This PCR is performed in 20 \textmu L reaction volumes using Taq polymerase (NEB \#M0285L) (again, for reasons of buffer compatibility) and 1 \textmu L MwoI (NEB \#R0573L) is added directly to the reaction mixture, which is then incubated at 60$^{\circ}$C for 1 hour. The reaction is then visualized on a 2-3\% agarose gel impregnated with SYBR Safe dye. In our hands, this second method is faster, easier, more robust, and more cost-effective. In both cases, the wildtype product is unable to be cut (single larger band) while the mutant is cleaved into two similarly sized smaller bands (a slightly hazy singleù lower band); the heterozygotes are differentiated by the presence of both bands. Confirmatory Sanger sequencing was performed as needed.

The allele nfatc3a\textsuperscript{xt59} was generated using an individual sgRNA (\seqsplit{5'Äô-TAATACGACTCACTATAGGGCAGTTTGCAGTAGTCATGTTTTAGAGCTAGAA-3'Äô}) and a mutation was identified containing a 22 bp deletion leading to a premature termination codon at the 8th amino acid (of 1074). The allele was identified by PCR amplification and Sanger sequencing using F: \seqsplit{5'Äô-GTTTTCCCAGTCACGACCAGAAGGTCGAGCAGTTTGG-3'} and R: \seqsplit{5'Äô-AACGTGTTTCGCCTTTGC-3'}. Sequencing used the "ÄúM13F(-40)"Äù primer supplied by the vendor (Eton Biosciences) (\seqsplit{5'Äô-GTTTTCCCAGTCACGAC-3'}). Genotyping was routinely conducted by high-resolution melt analysis (HRMA) using the MeltDoctor Master Mix (ThermoFisher \#4415450) with primers flanking the sgRNA site (\seqsplit{5'Äô-AAAGAGTCGGTGTACATAGACGGG-3'}Äô, \seqsplit{5''-CGAAGATCAGTCTGAAGTCCAGC-3'}). 

Crispant Assays 

To generate mosaic knockouts in genes of interest, we synthesized sgRNAs targeting the first exon of the respective genes. For nfatc2a we used \seqsplit{5'-TAATACGACTCACTATAGGTCAGTCAGGTGAACTGTCGTTTTAGAGCTAGAA-3'Äô} and for nfatc3a we used \seqsplit{5'Äô-TAATACGACTCACTATAGGTAGAGGCACTGACCTGCGGTTTTAGAGCTAGAA-3'Äô}. For prospective genotyping of these alleles, we used HRMA to assess approximate editing efficiency; this can only act as a rough proxy due to limitations and feasibility of exhausting genetic analysis of these mosaic larvae. For nfatc2a, we used the following primers: \seqsplit{5'Äô-CTCTTTTTACGGCGAAAAAGCTGC-3'ô}, \seqsplit{5'-GAAACAAACCTTGAAGTCCTGTTTGG-3'Äô}. For nfatc3a we used: \seqsplit{5'-AAAGAGTCGGTGTACATAGACGGG-3'}, \seqsplit{5'Äô-CGAAGATCAGTCTGAAGTCCAGC-3'}Äô. We had already begun generating the future stable alleles nfatc2a\textsuperscript{xt69} and nfatc3a\textsuperscript{xt59} and used these listed sgRNAs to increase our likelihood of introducing a functional mutation in these genes and to normalize target location and sgRNA number.


CLARITY + Microscopy

CLARITY fixation and clearing was conducted as previously described (Cronan et al., 2015). In brief, adult zebrafish were euthanized in tricaine, decapitated, and disemboweled. Visceral organs were immersed in an A1P4 CLARITY solution (4\% paraformaldehyde (EMS \#15710), 1\% acrylamide (Bio-Rad \#1610140), 0.05\% bis acrylamide (Bio-Rad \#1610142), 0.0025 g/ml radical initiator (Wako Chemical \#VA-044) in 1x final concentration PBS (Corning \#46013CM) and nutated at 4$^{\circ}$C for 2 days prior to overlay with mineral oil (Fisher Scientific \#BP2629) and polymerized at 37$^{\circ}$C for 3 hours. Hydrogel samples were collected, washed in 1x PBS, and then immersed in clearing solution at 37$^{\circ}$C (8\% sodium dodecyl sulfate (Bio-Basic \#SD8119) in 200 mM boric acid (Sigma-Aldrich \#B0394), pH 8.5), which was changed every 2-3 days until samples were optically clear. These samples were washed in 1x PBS supplemented to 0.1\% Triton-X (Fisher Scientific \#BP151) for two days at 37$^{\circ}$C with daily solution changes to remove excess SDS from the tissue. These tissues were then individually placed into black, opaque microcentrifuge tubes and immersed in refractive index matching solution (RIMS) (40 g, Histodenz (Sigma-Aldrich \#D2158), 30 mL 20 mM phosphate buffer (4.043 g Na2HPO4 (VWR \#BDH9296), 678.7 mg NaH2PO4 (Sigma-Aldrich \#S9638), 1 L diH2O), 0.01\% sodium azide (Sigma-Aldrich \#71290)) with rotation for at least 24 hours prior to imaging (Yang et al., 2014). 

Imaging was conducted on a spinning disk microscope (Zeiss AxioObserver Z1 connected to an XCite 120 LED Boost with an XLight 2TP, 89North LDI, Hamamatsu C13440 and captured on a Dell Precision Tower 5810 with Metamorph 7.10.5.476) in a MatTek dish (\#P35G-1.5-14-C) with optical bottom. Additional RIMS was added to the dish to cover the sample and minimize refraction during imaging. We panned across the proximal surface of the organ bundles to identify granulomas in each individual sample and captured Z-stack images of each of the identifiable granulomas at the maximum possible optical depth in the fish. This is able to capture the majority (but perhaps not all) of the granulomas present in a given fish due to inherent limitations in lens working distance.

All image processing was conducted in FIJI/ImageJ (Schneider et al., 2012). In-focus Z planes were identified and processed with the Maximum Intensity Projection function using a Jython macro. These files were saved and then subjected to cropping where the frame was cropped to the vasculature immediately surrounding each granuloma. This distance was unable to be precisely normalized across granulomas due to the differing sizes and shapes of the granulomas themselves as well as the nature of their varying physiological locations. Cropped images were then blinded using the blindrename.pl script (Salter, 2016). Images were then opened in ImageJ and vessels were traced using the segmented line tool, added to the Region of Interest (ROI) Manager tool and then measured for distance in pixels. Total length was then converted to microns based on the conversion factor provided by the microscope (1 px = 0.6552 \textmu m). Resulting .csv files were processed in Excel to remove unnecessary tag information from files names and then all subsequent analysis was performed in R using RStudio (citations).

qRT-PCR

THP-1 cells were transdifferentiated into macrophage-like cells using 50 ng/mL PMA (phorbol 12-myristate-13-acetate) (Sigma-Aldrich \#P148), seeded in 24 well cell culture treated plates at a concentration of 5 x 10\textsuperscript{5} cells/ml and incubated at 37$^{\circ}$C/5\%CO\textsubscript{2} for 48hr. After that the PMA media was changed using complete RPMI 1640 media and incubated at 37$^{\circ}$C/5\%CO\textsubscript{2} for 24hr (rest day). Then the cells were exposed to 0.5 mL of gamma-irradiated Mtb (BEI \#NR-49098) in 25\% glycerol (Sigma-Aldrich \#G7757) diluted in RPMI-1640 at a final concentration of 1 mg/mL. Cells were spun at 100 rcf for 5 min and incubated at 37$^{\circ}$C/5\%CO\textsubscript{2} for 8hr.

Cells then had media removed and were washed once with 1x PBS. After removing the PBS, 300 \textmu L of Trizol was add and cells were vigorously resuspended and moved into 1.5 mL microfuge tubes. RNA extraction was conducted by addition of 0.7 volumes of 1x TE (Sigma-Aldrich \#T9285) and 100 \textmu L of chloroform (EMD Millipore \#CX1055). After spinning at 17,000 rcf for 30 minutes at 4$^{\circ}$C, the upper aqueous layer was transferred to another tube, and 100 \textmu L of 24:1 chloroform:isoamyl alcohol (Sigma-Aldrich \#25666) was added. The tubes were then shaken by hand and spun for another 30 minutes at 17,000 rcf at 4$^{\circ}$C. The top aqueous layer was removed and final cleanup was done using the RNA Cleanup Kit (NEB \#T2040L) per the manufacturer'Äôs instructions.

cDNA synthesis was performed using the LunaScript RT SuperMix Kit (NEB \#E3010L) by the manufacturer'Äôs instructions. RT-PCR was performed using the Luna Universal qPCR Master Mix (NEB \#M3003L) in an Applied Biosystems 7500 Fast (ThermoFisher \#4351106) per the manufacturer's instructions. Final calculations were conducted in R. 

ELISA

Cells were cultured identically to previous, except they were plated in 96 well cell culture treated plates and exposed to gamma-irradiated \textit{M. tuberculosis} for a total of 24 hours to facilitate VEGF production and secretion. Supernatants were collected and spun down and then the upper layer was collected for further analysis. ELISA was performed according to the manufacturer's instructions (R\&D Systems \#DY293B). Absorbance was read on an Agilent Synergy LX plate reader.

Immunofluorescence

THP-1 cells were plated on 4-well chamber slides and differentiated with PMA at 50 ng/mL for 48 hours. Media was then replaced with fresh RPMI-1640 and cells were allowed to rest for 24 hours prior to further stimulation. Cells were then treated by addition of 1 mg/mL final concentration gamma-irradiated \textit{M. tuberculosis}, 40 \textmu M INCA-6 (Cayman Chemicals \#21812), and/or vehicle controls (25\% glycerol in PBS or DMSO (Fisher Scientific \#BP337), respectively). Cells were then incubated at 37$^{\circ}$C, 5\% CO2 for 8 hours and then fixed in 4\% PFA in 1x PBS for 20 minutes. Cells were then washed twice in 0.25\% NH\textsubscript{4}Cl (Sigma-Aldrich \#254134) or 0.15M glycine (to neutralize), rinsed in PBS, blocked in 2.5\% donkey serum (Fisher Scientific \#50413253) in 1x PBS for at least 20 minutes, and then incubated in primary antibody overnight at 4$^{\circ}$C. Cells were then rinsed, secondary antibody was added and cells were again incubated overnight at 4$^{\circ}$C. After 5x rinses in PBS, cells were dipped in diH\textsubscript{2}O and mounted in DAPI Fluoromount-G (SouthernBiotech \#0100-20), which was allowed to set overnight at RT in the dark. Slides were either stored at 4$^{\circ}$C in the dark prior to visualization or visualized immediately. 

Images shown in the figures were digitally adjusted for brightness and contrast in FIJI/ImageJ (Schindelin et al., 2012) and all adjustments were applied uniformly across the images within an experiment. All quantitation was performed based on the unadjusted brightness and contrast values and thresholded to better capture positive signal and eliminate the background fluorescence ubiquitous in these images.

Zeiss filter sets used were: 
\begin{itemize}
\item Filter Set 50 (Cy5, Alexa Fluor 647)
\item Filter Set 47 (CFP)
\item Filter Set 38 (GFP, Alexa Fluor 488)
\item Filter Set 43HE (tdTomato, Alexa Fluor 555)
\item Filter Set 46 (YFP)
\item Filter Set 49 (DAPI)
\end{itemize}

Lentivirus Construction

We sought to generate lentiviruses able to target multiple single guide RNAs to the same gene to maximize overall mutation rate and allow us to conduct experiments in mixed pools of heterogeneous cells, to minimize functional passage number. We therefore adopted a hybrid approach, inserting the sgRNA targeting array and hUbC promoter from Kabadi et al. 2014 (Addgene \#53190, a kind gift from Charles Gersbach) into the NotI/XbaI site of the lentiCRISPRv2 plasmid from Sanjana et al. 2014 (Addgene \#52961, a kind gift from Feng Zhang), creating a hybrid plasmid that simultaneously expressed Cas9, the puromycin resistance marker, and up to 4 single guide RNAs from a single plasmid.

Single guide RNA expression plasmids were cloned from phU6-gRNA, pmU6-gRNA, ph7SK-gRNA, and phH1-gRNA as described previous (Kabadi et al., 2014). The guide sequences for both NFATC2 and the safe targeting loci were chosen from the a database of available guides and safe loci in the human genome to model the DNA damage response from sgRNA targeting without overt toxicity or phenotypic changes (Morgens et al., 2017). These plasmids were purified and used in subsequent steps.

This resulting transfer empty vector (pLV hUbC-Cas9-P2A-Puro\_BsmBI-sgRNA-BsmBI, Addgene \#188703) was digested with Esp3I FastDigest (ThermoFisher \#FD0454) precisely as previously described (Kabadi et al., 2014) in the presence of equal masses (~200 ng each) of constituent sgRNA expression plasmids driven from mU6, hU6, 7SK, or hH1 RNA pol III promoters, ligated with T4 ligase (NEB \#M0202S), and cloned into NEB Stable (NEB \#C3040H) cells. Resulting plasmids were screened by restriction digestion and full plasmid sequencing. 

The appropriate lentivirus transfer plasmid was transfected into HEK293T cells alongside pMD2.G (Addgene \#12259) and psPAX2 (Addgene \#12260) (both kind gifts from Didier Trono) (plus sfGFP-C1 to mark transfected cells, Addgene \#54579, a kind gift of Michael Davidson \& Geoffrey Waldo) in a 4:3:1(:0.5) mass ratio using TransIT-Lenti reagents (Mirus Bio \#MIR-6603) (Pedelacq et al., 2006). Supernatants were collected 48 hours post transfection and immediately used to transduce THP-1 cells in the presence of 8 \textmu g/mL polybrene (Sigma-Aldrich \#TR-1003-G). Approximate titer was determined by infecting additional HEK293T cells with varying dilutions of the supernatant.

THP-1 Transduction

THP-1 cells were seeded in complete RPMI-1640 media supplemented with 8 \textmu g/mL polybrene (Sigma-Aldrich \#TR-1003-G) in two non-treated six-well plates at a concentration of 1 x 10\textsuperscript{6} cells/ml in each well. One six-well plate was infected with 1mL of pLV-ST and the other with pLV-NFATC2. The lentivirus infected THP-1 cells were spun at 1500 rcf/2 hr/22$^{\circ}$C, gently resuspended and incubated at 37$^{\circ}$C/5\%CO\textsubscript{2} for 72hr. Transduced cells were selected with 2 \textmu g/mL puromycin (Sigma-Aldrich \#P4512) for 48 hours and then kept in complete RPMI-1640 with 1 \textmu g/mL puromycin until time of assay.

Immunofluorescence analysis

To capture differences in VEGF expression across different experimental conditions, we programmatically blinded a subset of images from each experimental condition using blindrename.pl (Salter, 2016) and, using the \textit{ÄúCell Counter}Äù plugin in FIJI/ImageJ, we marked each nucleus (as a proxy for cell number), each cell that visually expressed VEGFA at a minimum/maximum bit value of 100/1500, cells that had nuclear translocation of NFATC2, cells at the intersection of these two factors, and, when applicable, Cas9 expression. These values were exported and subsequently processed in R.

\section{Quantification and Statistical Analysis}

All assays were performed under experimental blinding. For all assays where the genotype or experimental condition of the fish was apparent to the experimenter during data gathering (for instance, experiments in adults and the VIVIT assays in larvae), the resulting images were computationally blinded prior to analysis with either blindrename.pl (Salter, 2016) or an in-house Python translation (available at doi://10.5281/zenodo.6816429). For assays where the genotype is unknown (in-cross of heterozygotes experiments for card9, nfatc2a in the larvae), blinding was inherent in the design of the experiment and genotypes were matched to the individual fish post hoc. 

Statistical analysis was performed using R 4.2.1 within the latest version of RStudio IDE (R Core Team, 2022; RStudio Team, 2022). Graphing was performed using ggplot2 (Wickham, 2016; Wickham et al., 2022a). All statistical tests performed and the resulting significance values are indicated in figures and figure legends.

\section{R}

The R statistical environment began as a competing/complementary option to the existing S statistical programming language, but has become nearly ubiquitous in the life sciences over the past fifteen years or so. Offering new users an accessible environment in which to learn programming and with a large community, this language allows for reproducible, open-source analysis and visualizaiton of nearly any scientific data. R has become immensely flexible and libraries have been developed to accommodate any and every kind of numerical data. This also allow for community reanalysis and reinterpretation of existing data by facilitating the direct sharing of analysis methods and raw data. 

These features allowed me to develop a comprehensive set of infintely reproducible analysis scripts to analyze all of the data presented here. These scripts, along with the raw data, are available on Zenodo at doi://10.5281/zenodo.6816429. R (version 4.2.1, Äú\textit{Funny-Looking Kid}) was accessed via RStudio (\textit{Spotted Wakerobin}Äù version 2022.07.0) on macOS 12.6 Äú\textit{Monterey}Äù (R Core Team, 2022; RStudio Team, 2022). The libraries used to complete this work were dplyr (Wickham et al., 2022b), reshape (Wickham, 2022), and FSA (Ogle et al., 2022). Graphs utilized ggplot2 (Wickham, 2009; 2016; Wickham et al., 2022a), gghighlight (Yutani, 2022), ggbeeswarm (Clarke and Sherrill-Mix, 2017), ggsignif (Ahlmann-Eltze and Patil, 2021), scales (Wickham and Seidel, 2022), extrafont (Chang, 2022), and RColorBrewer (Neuwirth, 2022).

\section{FIJI/ImageJ}

FIJI/ImageJ is an extremely powerful image processing application developed initially by Wayne Rasband and now largely maintained by Curtis Reuden with support from the National Institutes of Health. Developed primarily in Java, this portable application has been instrumental in image analysis pipelines for over 20 years. Completely free and open-source, ImageJ as well as its successor ImageJ2 and the bundled plugins included with FIJI are the standard against which any competing applications are compared. Through the interpretation of images as numerical pixel values, ImageJ is able to perform nearly every conceivable operation on images to facilitate their revisualization and interpretation. Due to the numerical infrastructure of scientific imaging data, various aspects of these images can be distilled into various quantitative measurements, which is used throughout this work to analysis the angiogenesis phenotype. While the built-in capabilities of ImageJ have evolved over time, the primary advantage of using ImageJ is access to a robust programming environment based on Java in which to create plugins and macros in a range of Java-compatible programming languges, including Groovy, R, and Python as well as a beginner-friendly IJ1 macro language that, while limited, is able to be written trivially by new users. This work has necessitated the development of a range of these scripts and plugins, which have serviced the analysis of these images in the background both through automating repetitive processing tasks (Z projections, channel splitting, channel merging, file saving, lookup table application, etc.) and extracting quantitative data from these images in a high-throughput fashion. The details and functions of these scripts will be described in further detail in Chapter 4. 

Image analysis was conducted using the FIJI (Schindelin et al., 2012; Rueden et al., 2017) expansion of ImageJ (Girish and Vijayalakshmi, 2004; Schneider et al., 2012). Analysis pipelines were written in Jython (v.2.7.18) (van Rossum, 1995) and executed within the ImageJ Jython interpreter. All scripts are provided via Zenodo at the doi listed above.